### PR TITLE
Create bot to fetch images from a Discord channel of all threads

### PR DESCRIPTION
**Pull request by LuchoAravena**
-Updated the logic to retrieve images and text from a specific Discord channel instead of fetching data from the Notion database.
-Implemented new methods to fetch and process messages from Discord, ensuring compatibility with the existing functionality.

Closes [#103](https://github.com/kommitters/bas/issues/103)

### CODE REVIEW

File: lib/bas/bot/fetch_images_from_discord.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  class FetchImagesFromDiscord < Bot::Base
+    # read function to execute the PostgresDB Read component
+    #
+    def read
+      reader = Read::Default.new
+
+      reader.execute
+    end
+
+    # Process function to execute the Discord utility to fetch images from a discord channel threads
```
Comment (FelipeGuzmanSierra) [inline lib/bas/bot/fetch_images_from_discord.rb:46]: What do you think about changing it to `Process function to execute the Discord utility to fetch images from Discord channel threads.`? :thinking:

File: lib/bas/bot/write_media_review_in_discord.rb [hunk 2]
```diff
@@ -58,17 +54,15 @@ def read
       reader.execute
     end
 
-    # process function to execute the Notion utility to send formated blocks to a page
+    # process function to execute the Discord utility to send image feedback to a thread of Discord channel
```
Comment (FelipeGuzmanSierra) [inline lib/bas/bot/write_media_review_in_discord.rb:57]: What do you think about changing it to `process function to execute the Discord utility to send image feedback to a thread of a Discord channel.`? :thinking:

**Review by FelipeGuzmanSierra — CHANGES_REQUESTED** [review]
Great work! Just some comments

File: lib/bas/utils/discord/request.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      #
+      #
+      def self.execute(params)
+        url = URI.parse("#{DISCORD_BASE_URL}/#{params[:endpoint]}")
+        headers = headers(params[:secret_token])
+        response = HTTParty.send(params[:method], url, { headers: })
+        JSON.parse(response.body)
+      end
+
+      def self.get_thread_messages(params)
```
[the remark this answers is not in the document]
Comment (LuchoAravena) [reply lib/bas/utils/discord/request.rb:36]: We spoke about discard this comment to implement new logic of discord_bot

File: lib/bas/utils/discord/request.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        recent_messages = []
+        current_time = Time.now
+        messages = execute(params)
+        thread_ids = messages.select { |msg| msg["type"] == 18 }.map { |msg| msg["id"] }
```
[the remark this answers is not in the document]
Comment (LuchoAravena) [reply lib/bas/utils/discord/request.rb:40]: We spoke about discard this comment to implement new logic of discord_bot

File: lib/bas/utils/discord/request.rb [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+        return if thread_ids.empty?
+
+        thread_ids.each do |thread_id|
+          thread_messages = get_threads(thread_id, params)
+
+          thread_messages.each do |message|
+            message_time = Time.parse(message["timestamp"])
+
+            next unless (current_time - message_time) < 60 && message["attachments"]
```
[the remark this answers is not in the document]
Comment (LuchoAravena) [reply lib/bas/utils/discord/request.rb:50]: We spoke about discard this comment to implement new logic of discord_bot

File: lib/bas/utils/discord/request.rb [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            }
+          end
+        end
+        recent_messages
+      end
+
+      def self.get_threads(thread_id, params)
+        url = URI.parse("#{DISCORD_BASE_URL}/channels/#{thread_id}/messages")
+        headers = headers(params[:secret_token])
+        HTTParty.get(url, { headers: })
```
[the remark this answers is not in the document]
Comment (LuchoAravena) [reply lib/bas/utils/discord/request.rb:71]: Rubocop's suggested me to remove blank spaces

File: lib/bas/utils/discord/request.rb [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      end
+
+      def self.write_media_text(params)
+        url = URI.parse("#{DISCORD_BASE_URL}/#{params[:endpoint]}")
+        headers = headers(params[:secret_token])
+        HTTParty.send(params[:method], url, { body: params[:body].to_json, headers: })
```
[the remark this answers is not in the document]
Comment (LuchoAravena) [reply lib/bas/utils/discord/request.rb:77]: Same here, Rubocop suggested me to remove blank spaces

**Pull request merged**